static/app/views/dashboardsV2/detail.tsx

```diff
@@ -724,3 +697,49 @@ const StyledPageContent = styled(PageContent)`
 `;
 
 export default withApi(withOrganization(DashboardDetail));
+
+/**
+ * Reads the layout from an array of widgets.
+ */
+function getDashboardLayout(widgets: Widget[]): RGLLayout[] {
```

Comment:
nit: Should we put these layout related functions & constants in a layouts utils file to avoid clutter in details.tsx?

Reply:
Going to do this in a separate PR